tests/styles.spec.js

```diff
@@ -0,0 +1,55 @@
+const assert = require('assert');
+const { execSync } = require('child_process');
+const { mkdirSync, cpSync, readFileSync } = require('fs');
+
+/* eslint-disable */
+let copy;
+if (typeof cpSync === 'undefined') {
+  copy = require('recursive-copy');
+}
+/* eslint-enable */
+
+const { resolve } = require('path');
+const { cli, stdio } = require('./lib/cli');
+const {
+  copyRecursively,
+  createManifestFile,
+  createTemporaryWorkingDirectory,
+  deleteTemporaryWorkingDirectory,
+} = require('./lib/helpers');
+
+const basicStylesPath = resolve(__dirname, 'styles/samples/basic/scss');
+
+describe('Style tasks', () => {
+  describe('checks entire style task', () => {
+    const command = `${cli} styles`;
+    let cwd;
+    // Before test is ran, needs to compile styles into css.
+    before(async () => {
+      cwd = createTemporaryWorkingDirectory();
+      const tmpDirStylesPath = resolve(cwd, 'src/styles');
+      createManifestFile(cwd);
+      execSync('yarn add breakpoint-sass', { cwd, stdio });
+      mkdirSync(resolve(cwd, 'src'));
+      await copyRecursively(basicStylesPath, tmpDirStylesPath);
+      execSync(command, { cwd, stdio });
+    });
+
+    // after test is ran, delete that temporary compiled style.
+    after(() => deleteTemporaryWorkingDirectory(cwd));
+    it('styles task checked', () => {
+      const generatedFile = readFileSync(resolve(cwd, 'build/styles/styles-expanded.css').toString());
```

Comment:
@michelegrace I think it’s because of this little typo here. 😄 
```suggestion
      const generatedFile = readFileSync(resolve(cwd, 'build/styles/styles-expanded.css')).toString();
```

Reply:
😅 🙈 lol! thanks for catching that @agarzola

Reply:
My pleasure! It’s little wins like this that make my day. 😄 

BTW, I couldn’t help but peek at the most recent Windows failures. I wonder if the issue is line-endings. 🤔 The file generated during the test in Windows might have different line endings, so the assertion would fail when comparing the two. Might be worth looking into.

Reply:
Huh yeah, I see for the `unit-tests(12, windows-latest)` it looks like the file is printing with `\r\n` 🤔 Do I need some way to rewrite the control stylesheets if a user is using windows then and somehow check what system is being used prior to running the styles tests?

Reply:
Hmmm… I’m not sure what the solution is. What’s puzzling to me is that `\r\n` is the Windows line-ending and `\n` is Linux and `\r` is macOS. So why does the generated file (generated in Windows) have `\n` while the control files (generated in macOS) have `\r\n`.

That said, we might get around this with something like:
```javascript 
const controlFile = readFileSync(blablabla).toString().replace(/\r\n/g, '\n');
```

That would at least make the line endings the same for the sake of making the comparison, and any diffs that might result after that would be purely the result of differences in the actual CSS. What do you think?

Reply:
It'll give it a try! 🤞

Reply:
I applied the `replace()` to both the generated and control file. 🤔 And now all the tests are passing!

With that said, I currently just have two tests, one checking the expanded and minified stylesheets...does it make sense to go with the idea to test each pipeline separately (as much as possible), so perhaps the current `basic` sample could be simplified to test that `sassGlob()` works as expected, then another test sample with css properties that would allow testing `autoprefixer()` etc?

Reply:
@michelegrace I’m inclined to say we should keep these tests as simple as possible (i.e. the two all-encompassing tests you have) until we can say with confidence that having more granular testing would actually benefit us. Right now, with these tests you’ve written, if any upstream dependency update creates an unexpected difference in the way CSS files are generated, we will know about it. That is good enough for me.

What do you think?

Reply:
That works with me! I'll go ahead and write up the PR description etc and publish this for proper review then. 👍